pkgs/http/lib/src/io_client.dart

```diff
@@ -63,7 +67,24 @@ class IOClient extends BaseClient {
         ioRequest.headers.set(name, value);
       });
 
-      var response = await stream.pipe(ioRequest) as HttpClientResponse;
+      HttpClientResponse response;
+      if (onSendProgress != null) {
+        var loaded = 0;
+        onSendProgress(loaded, request.contentLength);
+
+        await ioRequest.addStream(
+          stream.map(
+            (chunk) {
+              loaded += chunk.length;
+              onSendProgress(loaded, request.contentLength);
+              return chunk;
+            },
+          ),
+        );
+        response = await ioRequest.close();
```

Comment:
Why to needs close this ioRequest

Reply:
It's been a while since I've last had a look at this.

But I need the response from the ioRequest for the code that follows and I guess that was the easiest way I could think of.

But if that is not a good idea or best practice I'm all in for improvements.